packages/editor/src/components/save-publish-panels/index.js

```diff
@@ -55,6 +57,49 @@ export default function SavePublishPanels( {
 		[]
 	);
 
+	const registry = useRegistry();
+
+	useEffect( () => {
+		addAction(
+			'editor.savePost',
+			'my-plugin/template-save-dialog',
+			async ( post, options ) => {
+				if ( options.isAutosave ) {
+					return;
+				}
+				if ( post.type !== 'wp_template' ) {
+					return;
+				}
+
+				const site = await registry
+					.select( coreStore )
+					.getEntityRecord( 'root', 'site' );
+				const template = await registry
+					.select( coreStore )
+					.getEditedEntityRecord( 'postType', post.type, post.id );
+				const editorSettings = await registry
+					.select( editorStore )
+					.getEditorSettings();
+
+				// Don't open for focused entity.
+				if ( editorSettings.onNavigateToPreviousEntityRecord ) {
```

Comment:
This is probably not the right fix to be honest. It's an indirect check. I'd be ok keeping it but I think the commend should be a lot more clear than that. we want to check "custom template". It's not out of the question that a regular template is also accessed as a focused entity, and also in the site editor you can also create custom templates without going through the page first.

Reply:
I'm surprised we don't have previous cases of "isHierarchyTemplate" or something like that.

Reply:
I don't think `isHierarchyTemplate` will help us tbh. It would be possible to have multiple post or page templates and edit them as a focussed entity here without it being _the_ active post/page template. While it needs to be possible to set it active generally, I don't think showing the activation panel in the focussed mode is expected.

Reply:
So in other words, the goal is not to check for "custom" templates, but rather if you're editing it outside of a page/post editor context.